mantle/kola/tests/misc/chronydhcp.go

```diff
@@ -0,0 +1,98 @@
+					  "verification": {}
+					},
+					"mode": 493
+				  }
+				]
+			  },
+			  "systemd": {
+				"units": [
+				  {
+					"contents": "[Unit]\nDescription=verify the DHCP propagation of NTP servers\n[Service]\nType=oneshot\nExecStart=/usr/local/bin/setup.sh\nRestart=on-failure\nRemainAfterExit=yes\n[Install]\nWantedBy=multi-user.target\n",
+					"enabled": true,
+					"name": "chrony-dhcp.service"
+				  }
+				]
+			  }
+			}`),
+	})
+}
+
+func verifyChronyDHCPSynchronization(c cluster.TestCluster) {
+	m := c.Machines()[0]
+	// containerID, err := c.SSH(m, "sudo podman run -d --privileged --network ns:/var/run/netns/container quay.io/sohankunkerkar/dnsmasq")
+	// if err != nil {
+	// 	c.Fatalf("Unable to start dnsmasq container: %v", err)
+	// }
+	// // Wait a little bit for the container to start
+	// if err := util.Retry(60, time.Second, func() error {
+	// 	// IP address of the NTP server is taken from https://tf.nist.gov/tf-cgi/servers.cgi
+	// 	// Name: time-c-g.nist.gov  IP address: 129.6.15.30
+	// 	cmd := fmt.Sprintf("sudo podman exec -it %s /bin/bash -c 'echo -e \"dhcp-range=172.16.0.10,172.16.0.20,12h\nbind-interfaces\ninterface=veth-container\ndhcp-option=option:ntp-server,129.6.15.30\" > /etc/dnsmasq.d/dhcp; sudo systemctl restart dnsmasq.service'", string(containerID))
+	// 	_, _, err = m.SSH(cmd)
+	// 	if err != nil {
+	// 		return err
+	// 	}
+	// 	return nil
+	// }); err != nil {
+	// 	c.Fatal("Unable to set up NTP server configuration: %v", err)
+	// }
+
+	// Wait a little bit for the chrony-dhcp.service to start
+	if err := util.WaitUntilReady(2*time.Minute, 10*time.Second, func() (bool, error) {
+		_, _, err := m.SSH("sudo systemctl status chrony-dhcp.service")
```

Comment:
I think you should disable the systemd unit by default, and then this code can just be:
`_, _, err := m.SSH("sudo systemctl start chrony-dhcp.service")`

That will avoid all race conditions between the service and kola connecting via ssh.